feat/Update plugin to expo 50

thomasrebam opened the pull request:
As Expo SDK 50 was published, we would need to be able to use this plugin, with the modifications that appear in this PR.

Indeed, as mentioned in the [Expo SDK 50 changelog](https://expo.dev/changelog/2024/01-18-sdk-50), they use React Native 0.73 version, where MainActivity file is not in Java anymore but in Kotlin. Therefore, changes in the withReactNativeKeyevent are needed if we want to use Expo 50.

thomasrebam committed:
Update to expo 50

silencer07 commented:
can we have this merged please? @thomasrebam did you by chance did an npm release for the mean time?

sergeushenecz commented:
When it will be merge. We wait it feature. Thanks for pr.

silencer07 commented:
@thomasrebam is there a chance to include the code to make the D_PAD work? I have the sample code here https://github.com/ChronSyn/react-native-keyevent-expo-config-plugin/issues/4

thomasrebam committed:
Adapt src file to new MainActivity

As Expo 50 is now using React Native 0.73 that uses MainActivity.kt and not MainActivity.java anymore

thomasrebam commented:
We published a patch [here](https://github.com/bamlab/react-native-keyevent-expo-config-plugin/tree/v1.0.51)

We didn't have time to check your patch @silencer07, so we did not add it to the fork.
